### PR TITLE
adding new retrieval endpoint

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/NEX-964

Introduce new endpoint to get shared stimulus.

p.s: Tests are being finished.

### CODE REVIEW

File: model/sharedStimulus/repository/SharedStimulusRepository.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+     * @throws core_kernel_classes_EmptyProperty
+     */
+    private function getContent(core_kernel_classes_Resource $resource): string
+    {
+        $link = $this->getPropertyValue($resource, MediaService::PROPERTY_LINK);
+
+        return (string)$this->getFileManagement()->getFileStream($link);
+    }
+
+    /**
```
Comment (siwane) [inline model/sharedStimulus/repository/SharedStimulusRepository.php:71]: Nice php doc ;p

Comment (gabrielfs7) [reply model/sharedStimulus/repository/SharedStimulusRepository.php:71]: True, I will remove the unnecessary part. :)

File: model/sharedStimulus/SharedStimulus.php [hunk 2]
```diff
@@ -54,6 +63,7 @@ public function jsonSerialize(): array
             'uri' => $this->uri,
             'languageUri' => $this->languageUri,
             'name' => $this->name,
+            'body' => str_replace(PHP_EOL, '', $this->body),
```
Comment (boajer) [inline model/sharedStimulus/SharedStimulus.php:66]: Naive question: is it safe to put raw XML into json property ( with no encoding, no escaping etc )

File: controller/SharedStimulus.php [hunk 3]
```diff
@@ -68,11 +93,21 @@ private function getCommandFactory(): CommandFactory
...
     {
         return $this->getServiceLocator()->get(CreateService::class);
     }
 
+    private function getSharedStimulusRepository(): SharedStimulusRepository
+    {
+        return $this->getServiceLocator()->get(SharedStimulusRepository::class);
+    }
+
     protected function getClassService()
```
Comment (mike-ionut-mihai-sandu-tao) [inline controller/SharedStimulus.php:111]: may i ask you to turn this to a private and add the response type

File: manifest.php [hunk 4]
```diff
@@ -28,7 +28,7 @@
     'label' => 'extension-tao-mediamanager',
     'description' => 'TAO media manager extension',
     'license' => 'GPL-2.0',
-    'version' => '9.5.0',
+    'version' => '9.6.0',
```
Comment (mike-ionut-mihai-sandu-tao) [inline manifest.php:31]: let's add the minimum required for the dependencies

File: model/sharedStimulus/FindQuery.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/sharedStimulus/FindQuery.php:20]: declare(strict_types=1);

File: model/sharedStimulus/SharedStimulus.php (hunk 2 above)
Comment (gabrielfs7) [reply model/sharedStimulus/SharedStimulus.php:66]: About security, if the user tries a XSS and submits a QTI with a malicious javascript core, we should check it in the input / import. I will check what we do, but then should we block all JS functionality? Btw, this is already outputted in the editInstance endpoint.

I tried to import this: 

```
<?xml version="1.0" encoding="UTF-8"?>
<div xmlns="http://www.imsglobal.org/xsd/imsqti_v2p2">
    <script>
        alert('TEST');
    </script>
</div>
```

And it is not allowed.

About the usability, the json_encode will escape the quotes, do FE will need to unescape this.

Do you suggest something?

File: controller/SharedStimulus.php (hunk 3 above)
Comment (gabrielfs7) [reply controller/SharedStimulus.php]: Method remove, cause was used in deprecated controller before. Thanks!

File: model/sharedStimulus/FindQuery.php (hunk 5 above)
Comment (gabrielfs7) [reply model/sharedStimulus/FindQuery.php:20]: Added to all new files. Thanks!

**Pull request merged**